File: 108_Convert_Sorted_Array_to_BST/108.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {

        return nums.size() > 0 ? buildBST(nums, 0, nums.size()-1) : nullptr ;
    }
    
private:
            TreeNode* buildBST(vector<int>& auxNums, int l, int r){
                if (l > r) return nullptr;
                int m = l + (r-l)/2;
                TreeNode* node = new TreeNode(auxNums[m]);
                node->left = buildBST(auxNums, l , m-1);
                node->right = buildBST(auxNums, m +1 , r);              
            return node;
        }
};

File: 108_Convert_Sorted_Array_to_BST/108.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:       
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        def buildBST(l:int,r:int) -> TreeNode:
            if l > r : return None
            m = l + (r-l) // 2
            node = TreeNode(nums[m] )
            node.left = buildBST(l, m-1)
            node.right = buildBST( m+1,r)
            return node
        # if len(nums) == 0: return None
        return buildBST(0, len(nums)-1) if len(nums) > 0 else None